plugins/scaffolder-backend/src/scaffolder/stages/publish/types.ts

```diff
@@ -29,11 +29,17 @@ export type PublisherBase = {
    *             catalog, plus the values from the form and the directory that has
    *             been templated
    */
-  publish(opts: {
-    entity: TemplateEntityV1alpha1;
-    values: RequiredTemplateValues & Record<string, JsonValue>;
-    directory: string;
-  }): Promise<{ remoteUrl: string }>;
+  publish(opts: PublisherOptions): Promise<PublisherResult>;
+};
+
+export type PublisherOptions = {
```

Comment:
I removed the `entity` parameter here since none of the publishers were using it. Maybe it needs to be added back in the future.